zerver/lib/rate_limiter.py

```diff
@@ -18,9 +18,13 @@
 client = get_redis_client()
 rules = settings.RATE_LIMITING_RULES
 def _rules_for_user(user):
-    # type: (UserProfile) -> Any
+    # type: (UserProfile) -> List[Tuple[int, int]]
     if user.rate_limits != "":
-        return [tuple(int(l) for l in limit.split(':')) for limit in user.rate_limits.split(',')] # type: List[Tuple[int, int]]
+        result = [] # type: List[Tuple[int, int]]
+        for limit in user.rate_limits.split(','):
+            (seconds, requests) = limit.split(':', 2)
+            result.append((int(seconds), int(requests)))
+        return result
```

Comment:
I looked at the combined diff on github instead of the individual commits. The commit messages explain everything. Sorry for the redundant question.